docs/src/main/tut/Job.md

```diff
@@ -1 +1,65 @@
-# Job
+# Job\[A\]
+
+On a `TypedDataset` all operations are lazy. An operation will either return a new 
+transformed `TypedDataset` or it will return a `Job[A]`, where A is the result of running a
+non-lazy computation in Spark. `Job` serves several functions: 
+
+- Makes all operations on a `TypedDataset` lazy, which gives less surprises compared to having
+few operations being lazy and other being strict
+- Allows the programmer to be more explicit when an expensive blocking operation needs to happen
+- Allows for Spark jobs to be lazily sequenced using monadic composition via for-comprehension
+- Provides an obvious place where you can annotate/name your Spark jobs to make it easier
```

Comment:
Meh I'm not really buying into this. If naming is needed in vanilla Spark, you could just put whatever into a `def`...